src/index.ts

```diff
@@ -56,11 +56,14 @@ if (args.patchall || args.patch) {
     ['SIGINT', 'SIGTERM', 'SIGQUIT'].forEach((signal: NodeJS.Signals) =>
         process.on(signal, () => {
             log("info", "Shutting down...");
-            client.destroy();
-            webServer.close();
-            database.close();
-            log("info", "Bye");
-            process.exit(0);
+            client.prepareShutdown()
```

Comment:
Personally, I prefer the `await`-syntax as I find it more readable. But that is mostly personal preference.

Reply:
this is not an async function and therefore there is no `await` :)